hooks/pre-push

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+#
+# Git pre-push hook to validate that tag versions match the Version constant in version.go
+#
+# This hook is called with the following parameters:
+# $1 -- Name of the remote to which the push is being done
+# $2 -- URL to which the push is being done
+#
+# Information about the commits which are being pushed is supplied as lines to
+# the standard input in the form:
+#   <local ref> <local sha1> <remote ref> <remote sha1>
+#
+
+set -euo pipefail
+
+print_error() {
+    echo "ERROR: $1" >&2
+}
+
+print_warning() {
+    echo "WARNING: $1" >&2
+}
+
+print_info() {
+    echo "INFO: $1" >&2
+}
+
+# Function to extract version from version.go
+get_version_from_file() {
+    if [[ ! -f "version.go" ]]; then
```

Comment:
This func assumes we are in the root of the directory :thinking:  should we validate that to either include a better error message, or some traversal to get to the root